pp_ctl.c

```diff
@@ -4792,6 +4787,36 @@ PP(pp_leaveeval)
     else
         leave_adjust_stacks(oldsp, oldsp, gimme, 0);
 
+    /* did require return a false value?
+       make sure to initialize failed, whatever circumstance this was called
+       but check if use feature 'module_true' is enabled where it matters
+     */
+    if (CxOLD_OP_TYPE(cx) == OP_REQUIRE) {
+        COP *old_pl_curcop = PL_curcop;
+        if (PL_op->op_flags & OPf_KIDS && OP_TYPE_IS_OR_WAS(PL_op, OP_LEAVEEVAL)) {
+            const OP *kid = cLISTOPx(cUNOPx(PL_op)->op_first)->op_first;
+
+            for (; kid; kid = OpSIBLING(kid)) {
+                if (OP_TYPE_IS_OR_WAS(kid, OP_NEXTSTATE)) {
+                    PL_curcop = cCOPx(kid);
+                    module_true = FEATURE_MODULE_TRUE_IS_ENABLED;
+                }
+            }
+        }
+        PL_curcop = old_pl_curcop;
+
+        /* failed stays false */
```

Comment:
I'd bring the `failed` variable down here. Start with

```c
bool failed;
```

then inside the true branch, put `failed = false` so it's clear both sides of the if will assign a value into it.

Reply:
Also done.